src/pages/webhooks/use-cases.md

```diff
@@ -0,0 +1,370 @@
+                    <field name="giftCard.gift_cards" source="giftCardAccountData.gift_cards" />
+                </fields>
+            </hook>
+        </batch>
+    </hooks>
+</method>
+```
+
+**Configured payload:**
+
+```json
+{
+   "giftCard" {
+        "cartId": null,
+        "gift_cards": "string[]"
+    }
+}
+```
+
+**Endpoint code example:**
+
+```js
+const fetch = require('node-fetch')
+const { Core } = require('@adobe/aio-sdk')
+const { errorResponse, getBearerToken, stringParameters, checkMissingRequestInputs } = require('../utils')
+ 
+// main function that will be executed by Adobe I/O Runtime
+async function main (params) {
+  // create a Logger
+  const logger = Core.Logger('main', { level: params.LOG_LEVEL || 'info' })
+ 
+  try {
+    // 'info' is the default level if not set
+    logger.info('Calling the main action')
+ 
+    // log parameters, only if params.LOG_LEVEL === 'debug'
+    logger.debug(stringParameters(params))
+ 
+    // check for missing request input parameters and headers
+    const requiredParams = [/* add required params */]
+    const requiredHeaders = ['Authorization']
+    const errorMessage = checkMissingRequestInputs(params, requiredParams, requiredHeaders)
+    if (errorMessage) {
+      // return and log client errors
+      return errorResponse(400, errorMessage, logger)
+    }
+ 
+    // Place the real validation (calling 3rd party endpoints) here.
+    // In this example, we check if the postal code is larger than 70000.
+    // If it is, the request is considered invalid.
+
+    const response = {statusCode: 200}
+    const address = params.address
+    if (address.postcode > 70000) {
+      response.body = JSON.stringify({
+        op: "exception",
+        message: `App Builder Webhook Response: The address with postcode  "${address.postcode}" is not valid`,
+        type: "Magento\\Framework\\Exception\\InputException"
+      })
+    } else {
+      response.body = JSON.stringify({
+        op: "success"
+      })
+    }
+ 
+    return response
+  } catch (error) {
+    // log any server errors
+    logger.error(error)
+    // return with 500
+    return errorResponse(500, 'server error', logger)
+  }
+}
+ 
+exports.main = main
+```
+
+If validation fails, the runtime AppBuilder action returns an exception message. The message is visible to the customer.
+
+```js
+response.body = JSON.stringify({
+    op: "exception",
+    message: `App Builder Webhook Response: The gift card code "${giftCards[i]}" is not valid`
+})
+```
+
+## Add product to cart
+
+When a shopper adds a product to the cart, a third-party inventory management system checks whether the item is in stock. If it is, allow the product to be added. Otherwise, display an error message.
+
+**Webhook:**
+
+`observer.checkout_cart_product_add_before`
+
+**Default Payload:**
+
+The `observer.checkout_cart_product_add_before` webhook is not based on the resource model. The payload is obtained from the code execution in the application.
```

Comment:
`webhook is not based on the resource model` I think it's better not to mention it as in the payload we can see the converted product object (Resource Model) as one of the arguments.
I mentioned that previously as we can't automatically get payload info if the observer event is not called inside the resource module (like afterSave, afterDelete).
Maybe we can rephrase it to something like:
```
The example of the `observer.checkout_cart_product_add_before` payload obtained from the code execution in the application.
```